code/day2/4_act_data_in_actel.R

```diff
@@ -0,0 +1,218 @@
+
+# Tack OTN stations to the end of the MATOS extract.
+# These are in the exact same format because OTN and MATOS's databases are the
+# same format, so we can easily coordinate our output formats.
+all_stations <- bind_rows(deploy_metadata, otn_deploy_metadata)
+
+# For ACT/FACT projects - we could use GeoServer to share this information, and could even add
+# an authentication layer to let ACT members do this same trick to fetch deploy histories!
+
+# So now this is our animal tagging metadata
+# tag_metadata %>% View
+
+# these are our detections: 
+
+# proj_dets %>% View
+
+# These are our deployments:
+
+# all_stations %>% View
+
+# Mutate metadata into Actel format ----
+
+# Create a station entry from the glatos array and station number.
+# --- add station to receiver metadata ----
+full_receiver_meta <- all_stations %>%
+  dplyr::mutate(
+    station = paste(collectioncode, station_name, sep = '-')
+  ) %>% 
+  filter(is.na(rcvrstatus)|rcvrstatus != 'lost')
+
+# Actel Biometrics ------------
+
+# All dates will be supplied to Actel in this format:
+actel_datefmt = '%Y-%m-%d %H:%M:%S'
+
+# biometrics is the tag metadata. If you have a tag metadata sheet, it looks like this:
+actel_biometrics <- tag_metadata %>% dplyr::mutate(Release.date = format(UTC_RELEASE_DATE_TIME, actel_datefmt),
+                                    Signal=as.integer(TAG_ID_CODE),
+                                    Release.site = RELEASE_LOCATION, 
+                                    # Group=RELEASE_LOCATION  # can supply group to subdivide tagging groups
+                                    )
+# Actel Deployments ----
+# deployments is based in the receiver deployment metadata sheet
+actel_deployments <- full_receiver_meta %>% dplyr::filter(!is.na(recovery_date)) %>%
+  mutate(Station.name = station,
+         Start = format(deploy_date, actel_datefmt), # no time data for these deployments
+         Stop = format(recovery_date, actel_datefmt),  # not uncommon for this region
+         Receiver = rcvrserial) %>%
+  arrange(Receiver, Start)
+
+
+# Actel Detections -------------------
+
+# Renaming some columns in the Detection extract files   
+actel_dets <- proj_dets %>% dplyr::mutate(Transmitter = tagname,
+                                    Receiver = as.integer(receiver),
+                                    Timestamp = format(datecollected, actel_datefmt),
+                                    CodeSpace = extractCodeSpaces(tagname),
+                                    Signal = extractSignals(tagname))  %>%
+                            dplyr::filter(Receiver != 'Release')
+                      
+# We don't have any environmental data in our detection extract here, but Actel 
+# will also find and plot temperature or other sensor values if you have those
+# kinds of tags.
+
+# Actel Receivers-------------------
+
+# Prepare and style entries for receivers
+actel_receivers <- full_receiver_meta %>% dplyr::mutate( Station.name = station,
```

Comment:
got a warning from running this part
```
Warning message:
Problem with `mutate()` input `Receiver`.
i NAs introduced by coercion
i Input `Receiver` is `as.integer(receiver)`. 
```

Reply:
fixed this by reversing the filter and mutate steps to avoid trying to filter 'release' detections.